file-service/src/test/java/com/extremecoder/fileservice/controller/FileControllerTest.java

```diff
@@ -0,0 +1,118 @@
+package com.extremecoder.fileservice.controller;
+
+import com.extremecoder.fileservice.model.FileInfo;
+import com.extremecoder.fileservice.service.FileStorageService;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * File Controller Test
+ *
+ * @author Badrul
+ */
+
+@Slf4j
+@ExtendWith(MockitoExtension.class)
+class FileControllerTest {
+
+    @InjectMocks
+    FileController fileController;
+
+    @Mock
+    FileStorageService fileStorageService;
+
+    static MockMultipartFile sampleFile1;
+
+    static MockMultipartFile sampleFile2;
+
+    static String fileName1 = "sampleFile1.txt";
+    static String fileName2 = "sampleFile2.txt";
+
+    @BeforeAll
+    static void beforeAll() {
+        log.info("*** FileControllerTest.beforeAll()***");
+
+        sampleFile1 = new MockMultipartFile(fileName1, fileName1, "text/plain", "Hello World! from 1".getBytes());
+
+        sampleFile2 = new MockMultipartFile(fileName2, fileName2, "text/plain", "Hello World! from 2".getBytes());
+
+    }
+
+    @Test
+    @DisplayName("when uploadFile() is called then upload single file")
+    void uploadFile() {
+        log.info("*** FileControllerTest.uploadFile() -- started -- ***");
+
+
+        FileInfo fileInfo = FileInfo.builder().fileName(fileName1).originalFilename(fileName1).build();
```

Comment:
Things should be grouped:
Arrange -> Where all the dependencies are arranged like: line number: 61,67
Act -> Where we stub the specific methods using `when..then` block
Assert -> Where we test our final results
please change accordingly in all the test cases.